File: 品味Java的21種設計模式/dp_cpp/decorator/example2/C2.cpp
#include "C2.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace decorator
			{
				namespace example2
				{

					void C2::a1()
					{
						//转调A对象的功能
						a->a1();
					}

					void C2::c11()
					{
						puts("now in C2.c11");
					}
				}
			}
		}
	}
}

File: 品味Java的21種設計模式/dp_java/decorator/example2/C2.java
package cn.javass.dp.decorator.example2;

public class C2 {
    /**
     * 创建A对象的实例
     */
    private A a = new A();

    public void a1() {
        //转调A对象的功能
        a.a1();
    }
    public void c11() {
        System.out.println("now in C2.c11");
    }
}
